File: python/1118.py
response = 1
grade1 = -1
grade2 = -1

while response == 1:
    while True:
        grade1 = float(input())
        if grade1 < 0 or grade1 > 10:
            print('nota invalida')
        else:
            break
    while True:
        grade2 = float(input())
        if grade2 < 0 or grade2 > 10:
            print('nota invalida')
        else:
            break
        
    avg = (grade1 + grade2) / 2
    print(f'media = {avg:.2f}')
    while True:
        print('novo calculo (1-sim 2-nao)')
        response = int(input())
        if response == 2 or response == 1:
            break

File: lua/1118.lua
local continue = 1
while continue == 1 do
    local count = 0
    local sum = 0
    while count < 2 do
        local n = io.read("n")
        if n >= 0 and n <= 10 then
            count = count + 1
            sum = sum + n
        else
            print('nota invalida')
        end
    end
    print(string.format("media = %.2f", sum / 2))
    repeat
        print("novo calculo (1-sim 2-nao)")
        continue = io.read('n')
    until continue == 1 or continue == 2
end